agents-run-api/src/agents/generateTaskHandler.ts

```diff
@@ -111,6 +112,29 @@ export const createTaskHandler = (
       ]);
 
       logger.info({ toolsForAgent, internalRelations, externalRelations }, 'agent stuff');
+      
+      // Enhance internal relation descriptions with their transfer/delegation info
+      const enhancedInternalRelations = await Promise.all(
+        internalRelations.map(async (relation) => {
+          try {
+            const relatedAgent = await getAgentById(dbClient)({
+              scopes: { tenantId: config.tenantId, projectId: config.projectId },
+              agentId: relation.id,
+            });
+            if (relatedAgent) {
+              const enhancedDescription = await generateDescriptionWithTransfers(
```

Comment:
I see this is called in two places? once in genreateTaskHandler and another time when making the agent card. Can we reduce the places it is called to one?